icu4j/main/classes/core/src/com/ibm/icu/impl/number/parse/NumberParserImpl.java

```diff
@@ -146,6 +146,7 @@ public static NumberParserImpl createParserFromProperties(
         }
         Currency currency = CustomSymbolCurrency.resolve(properties.getCurrency(), locale, symbols);
         boolean isStrict = properties.getParseMode() == ParseMode.STRICT;
+        boolean isCompatibilityMode = properties.getParseMode() == ParseMode.COMPATIBILITY;
```

Comment:
Optional: make a variable `ParseMode parseMode = properties.getParseMode()` and then change all the if statements to just operate in terms of a single equality operation on the parseMode:

```java
if (isStrict) { // old
if (parseMode == ParseMode.STRICT) { //new

if (isStrict || isCompatibilityMode) { // old
if (parseMode != ParseMode.LENIENT) { // new
```